cmd/cloudFoundryCreateService.go

```diff
@@ -0,0 +1,115 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/SAP/jenkins-library/pkg/cloudfoundry"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+func cloudFoundryCreateService(config cloudFoundryCreateServiceOptions, telemetryData *telemetry.CustomData) {
+
+	cf := cloudfoundry.CFUtils{Exec: &command.Command{}}
+
+	err := runCloudFoundryCreateService(&config, telemetryData, cf)
+
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+
+}
+
+func runCloudFoundryCreateService(config *cloudFoundryCreateServiceOptions, telemetryData *telemetry.CustomData, cf cloudfoundry.CFUtils) error {
+
+	var err error
+
+	var c = cf.Exec
+
+	loginOptions := cloudfoundry.LoginOptions{
+		CfAPIEndpoint: config.CfAPIEndpoint,
+		CfOrg:         config.CfOrg,
+		CfSpace:       config.CfSpace,
+		Username:      config.Username,
+		Password:      config.Password,
+	}
+
+	err = cf.Login(loginOptions)
+
+	if err != nil {
+		return fmt.Errorf("Error while logging in: %w", err)
+	}
+
+	if err == nil {
```

Comment:
The if condition here is not needed. When reaching this line of code we already know that `err` is `nil` (return above).

Reply:
True, removed the if condition